packages/driver/test/cypress/integration/commands/actions/check_spec.js

```diff
@@ -0,0 +1,1233 @@
+    })
+
+    it('waits until element is no longer disabled', () => {
+      const chk = $(':checkbox:first').prop('disabled', true)
+
+      let retried = false
+      let clicks = 0
+
+      chk.on('click', () => {
+        clicks += 1
+      })
+
+      cy.on('command:retry', _.after(3, () => {
+        chk.prop('disabled', false)
+        retried = true
+      }))
+
+      cy.get(':checkbox:first').check().then(() => {
+        expect(clicks).to.eq(1)
+
+        expect(retried).to.be.true
+      })
+    })
+
+    it('delays 50ms before resolving', () => {
+      cy.$$(':checkbox:first').on('change', (e) => {
+        cy.spy(Promise, 'delay')
+      })
+
+      cy.get(':checkbox:first').check().then(() => {
+        expect(Promise.delay).to.be.calledWith(50, 'click')
+      })
+    })
+
+    describe('assertion verification', () => {
+      beforeEach(() => {
+        cy.on('log:added', (attrs, log) => {
+          if (log.get('name') === 'assert') {
+            this.lastLog = log
+          }
+        })
+      })
+
+      it('eventually passes the assertion', () => {
+        cy.$$(':checkbox:first').click(function () {
+          _.delay(() => {
+            cy.$$(this).addClass('checked')
+          }
+          , 100)
+        })
+
+        cy.get(':checkbox:first').check().should('have.class', 'checked').then(() => {
+          const {
+            lastLog,
+          } = this
+
+          expect(lastLog.get('name')).to.eq('assert')
+          expect(lastLog.get('state')).to.eq('passed')
+
+          expect(lastLog.get('ended')).to.be.true
+        })
+      })
+
+      it('eventually passes the assertion on multiple :checkboxs', () => {
+        cy.$$(':checkbox').click(function () {
+          _.delay(() => {
+            cy.$$(this).addClass('checked')
+          }
+          , 100)
+        })
+
+        cy.get(':checkbox').invoke('slice', 0, 2).check().should('have.class', 'checked')
+      })
+    })
+
+    describe('events', () => {
+      it('emits click event', (done) => {
+        $('[name=colors][value=blue]').click(() => {
+          done()
+        })
+
+        cy.get('[name=colors]').check('blue')
+      })
+
+      it('emits change event', (done) => {
+        $('[name=colors][value=blue]').change(() => {
+          done()
+        })
+
+        cy.get('[name=colors]').check('blue')
+      })
+
+      it('emits focus event', () => {
+        let focus = false
+
+        $('[name=colors][value=blue]').focus(() => {
+          focus = true
+        })
+
+        cy.get('[name=colors]')
+        .check('blue')
+        .then(() => {
+          expect(focus).to.eq(true)
+        })
+      })
+    })
+
+    describe('errors', () => {
+      beforeEach(() => {
+        Cypress.config('defaultCommandTimeout', 100)
+
+        this.logs = []
+
+        cy.on('log:added', (attrs, log) => {
+          this.lastLog = log
+
+          this.logs.push(log)
```

Comment:
`this` with arrow functions, is this maintaining the correct `this`?

Reply:
Yes, the `cy.on('fail', (err) => ...` in the subsequent tests receives this.logs